File: src/eu/rafig/covid/core/game/entities/specs/Vitals.java
package eu.rafig.covid.core.game.entities.specs;

import com.sun.xml.internal.bind.v2.runtime.reflect.opt.Const;
import eu.rafig.covid.core.game.common.Constants;

public class Vitals {
    private Health health ;
    private Energy energy;
    private Money money;

    public Vitals() {
        health = new Health(Constants.STARTING_HEALTH);
        money = new Money(Constants.STARTING_MONEY);
        energy = new Energy(Constants.STARTING_ENERGY);
    }

    public Health getHealth() {
        return health;
    }

    public Money getMoney() {
        return money;
    }

    public Energy getEnergy() {
        return energy;
    }
}

File: Assets/src/C#/entities/specs/Vitals.cs
using System.Collections;
using System.Collections.Generic;
using eu.parada.common;

namespace eu.parada.entities.specs {
    public class Vitals {
        public Health health { get; private set; }
        public Energy energy { get; private set; }
        public Money money { get; private set; }

        public Vitals() {
            health = new Health(Constants.STARTING_HEALTH);
            money = new Money(Constants.STARTING_MONEY);
            energy = new Energy(Constants.STARTING_ENERGY);
        }
        public Vitals(int difficulty) {
            health = new Health(Constants.STARTING_HEALTH - (difficulty/ Constants.VITALS_DIFFICULTY_CONSTANT));
            energy = new Energy(Constants.STARTING_ENERGY - (difficulty/ Constants.VITALS_DIFFICULTY_CONSTANT));
            money = new Money(Constants.STARTING_MONEY);
        }
    }
}
